[BUG] `LockFailedException` when creating a new password

**Describe the bug**
I have somehow gotten my repositories (both the Play Store and a dev version) into a state where creating a new password leads to the following exception:

```
GitCommandExecutor$execute: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of add command
GitCommandExecutor$execute: 	at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:48)
GitCommandExecutor$execute: 	at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:1)
GitCommandExecutor$execute: 	at com.zeapo.pwdstore.git.GitCommandExecutor$execute$3.invokeSuspend(GitCommandExecutor.kt:2)
GitCommandExecutor$execute: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
GitCommandExecutor$execute: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:15)
GitCommandExecutor$execute: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
GitCommandExecutor$execute: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:13)
GitCommandExecutor$execute: Caused by: org.eclipse.jgit.errors.LockFailedException: Cannot lock /data/user/0/dev.msfjarvis.aps/files/store/.git/index
GitCommandExecutor$execute: 	at org.eclipse.jgit.lib.Repository.lockDirCache(Repository.java:19)
GitCommandExecutor$execute: 	at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:6)
GitCommandExecutor$execute: 	... 6 more
```

This is not fixed by deleting and re-cloning the repository and likely caused by `.git/index.lock` sticking around after another Git operation.

Additionally, the error dialog box that accompanies the exception is only shown briefly and there is a warning about a leaked `DecorView`, which hints at some logic involving `finish` and errors being incorrect.

**To Reproduce**
Steps to reproduce the behavior:
1. Cause some unknown corruption during/after a Git operation.
2. Create a password, which triggers a git add.

**Expected behavior**
The password is added and committed.

As a bandaid fix, we could potentially delete `.git/index.lock` e.g. when the app starts. But we should see whether we can reproduce the underlying repo corruption.

**Screenshots**
N/A

**Device information (please complete the following information):**
 - Device: Google Pixel 3a
 - OS: Android 11 Beta 3
 - App version: 1.11.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ All notable changes to this project will be documented in this file.
 ### Fixed
 
 -   Password creation UI will scroll if it does not fit on the screen
+-   Saving a password after creating it fails to finish commit operation
 
 ## [1.11.1] - 2020-08-21
 

--- a/app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt
+++ b/app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt
@@ -440,13 +440,15 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
                                     returnIntent.putExtra(RETURN_EXTRA_USERNAME, username)
                                 }
 
-                                lifecycleScope.launch {
-                                    commitChange(
-                                        getString(
-                                            R.string.git_commit_edit_text,
-                                            getLongName(fullPath, repoPath, editName)
+                                if (editing) {
+                                    lifecycleScope.launch {
+                                        commitChange(
+                                            getString(
+                                                R.string.git_commit_edit_text,
+                                                getLongName(fullPath, repoPath, editName)
+                                            )
                                         )
-                                    )
+                                    }
                                 }
 
                                 if (directoryInputLayout.isVisible && directoryInputLayout.isEnabled && oldFileName != null) {

